sqlite3.go

```diff
@@ -2083,6 +2083,12 @@ func (s *SQLiteStmt) execSync(args []driver.NamedValue) (driver.Result, error) {
 		return nil, err
 	}
 
+	rv = C.sqlite3_reset(s.s)
+	if rv != C.SQLITE_OK {
```

Comment:
Per the sqlite documentation ([ref](https://www.sqlite.org/c3ref/reset.html)),
> If the most recent call to [sqlite3_step(S)](https://www.sqlite.org/c3ref/step.html) for the [prepared statement](https://www.sqlite.org/c3ref/stmt.html) S indicated an error, then [sqlite3_reset(S)](https://www.sqlite.org/c3ref/reset.html) returns an appropriate [error code](https://www.sqlite.org/rescode.html). The [sqlite3_reset(S)](https://www.sqlite.org/c3ref/reset.html) interface might also return an [error code](https://www.sqlite.org/rescode.html) if there were no prior errors but the process of resetting the prepared statement caused a new error. For example, if an [INSERT](https://www.sqlite.org/lang_insert.html) statement with a [RETURNING](https://www.sqlite.org/lang_returning.html) clause is only stepped one time, that one call to [sqlite3_step(S)](https://www.sqlite.org/c3ref/step.html) might return SQLITE_ROW but the overall statement might still fail and the [sqlite3_reset(S)](https://www.sqlite.org/c3ref/reset.html) call might return SQLITE_BUSY if locking constraints prevent the database change from committing. Therefore, it is important that applications check the return code from [sqlite3_reset(S)](https://www.sqlite.org/c3ref/reset.html) even if no prior call to [sqlite3_step(S)](https://www.sqlite.org/c3ref/step.html) indicated a problem.

that is why we consider the execution failed if `reset` returns an error code.